android/src/com/frostwire/android/gui/fragments/BrowsePeerFragment.java

```diff
@@ -485,6 +688,114 @@ private void changeSelectedRadioButton(byte currentFileType, byte nextFileType)
             nextButton.setChecked(true);
             nextButton.callOnClick();
         }
+
+    }
+
+    public boolean inSpecialState() {
+        return fragmentState != ViewState.NORMAL;
+    }
+
+    public void endSpecialState() {
+        if (fragmentState == ViewState.CHECK) {
+            endCheckState();
+        } else if (fragmentState == ViewState.FILTERING) {
+            endFilteringState();
+        } else if (fragmentState == ViewState.FILTERED) {
+            endFilteredState();
+        }
+    }
+
+    public void endCheckState() {
+        fragmentState = beforeCheckState;
+        if (beforeCheckState == ViewState.FILTERED) {
+            ((MainActivity) getActivity()).swapDrawerForBack();
+        }
+        swipe.setOnSwipeListener(swipeListener);
+        adapter.setCheckState(false);
+        updateViewsToState();
+    }
+
+    public void enterCheckState() {
```

Comment:
to keep code consistent I believe this method should go above `endCheckState()` all the other ones you have, enterXXXState first endXXXState second.